chef/cookbooks/nova/recipes/compute.rb

```diff
@@ -130,19 +130,41 @@ def set_boot_kernel_and_trigger_reboot(flavor='default')
   end
 
   case node[:nova][:libvirt_type]
-    when "kvm"
-      package "kvm"
+    when "kvm", "qemu"
+      if node.platform_version.to_f >= 12.0
+        package "qemu"
+        package "qemu-kvm"
+        package "qemu-block-rbd"
+      else
+        package "kvm"
+      end
+
+      # Install Ceph integration if needed
+      cinder_servers = search(:node, "roles:cinder-controller") || []
```

Comment:
Hrm, shouldn't that be `search_env_filtered(:node, "roles:cinder-controller")`?

Also, we have a similar code in `config.rb`, so maybe we should move that code there? (Or well, probably better to do it the other way around and move the `config.rb` code here...)